packages/core-data/src/test/integration.js

```diff
@@ -38,10 +34,13 @@ const runPromise = async ( promise ) => {
 };
 
 const runPendingPromises = async () => {
-	jest.runAllTimers();
-	const p = new Promise( ( resolve ) => setTimeout( resolve ) );
-	jest.runAllTimers();
-	await p;
+	// @TODO: find a better way of exhausting the current event loop queue
+	for ( let i = 0; i < 100; i++ ) {
+		jest.runAllTimers();
+		const p = new Promise( ( resolve ) => setTimeout( resolve ) );
+		jest.runAllTimers();
+		await p;
+	}
```

Comment:
I have a hard time understanding what is this integration test testing. Instead of calling `jest.runAllTimers()` all the time, can we use `jest.useRealTimers()` and properly await a relevant promise before asserting some fact? That's a suggestion for a rewrite in another PR. For now, hacking our way through the test is probably OK.